scripts/helpers/customBuild.js

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+const fs = require("fs");
+const path = require("path");
+const { rollup } = require("rollup");
+const nodeResolve = require("@rollup/plugin-node-resolve").default;
+const commonjs = require("@rollup/plugin-commonjs");
+const babel = require("@rollup/plugin-babel").default;
+const terser = require("rollup-plugin-terser").terser;
+const yargs = require("yargs/yargs");
+const { hideBin } = require("yargs/helpers");
+const conditionalBuildBabelPlugin = require("./conditionalBuildBabelPlugin");
+
+// Path to componentCreators.js
+const componentCreatorsPath = path.join(
+  __dirname,
+  "../../src/core/componentCreators.js"
+);
+
+// Read componentCreators.js
+const componentCreatorsContent = fs.readFileSync(componentCreatorsPath, "utf8");
+
+// Extract required components from componentCreatorsContent
+const requiredComponents = componentCreatorsContent
+  .match(/REQUIRED_COMPONENTS = \[[\s\S]*?\]/)[0]
+  .match(/create[A-Z]\w+/g)
+  .map(component => component.slice(6).toLowerCase());
```

Comment:
What does the `6` represent? I would like to see this done in a way that doesn't have a (seemingly) random number. Same thing with the one on line 41.